core/src/main/java/org/apache/druid/data/input/impl/MapInputRowParser.java

```diff
@@ -69,29 +66,32 @@ public static InputRow parse(InputRowSchema inputRowSchema, Map<String, Object>
     return parse(inputRowSchema.getTimestampSpec(), inputRowSchema.getDimensionsSpec(), theMap);
   }
 
-  private static InputRow parse(
-      TimestampSpec timestampSpec,
+  private static List<String> findDimensions(
```

Comment:
Maybe add some comment here on 
- how we are using the flag and fields to include/exclude dimensions
- what is `theMap` expected to contain (I guess we could use a better name for this field too)

Reply:
I added a javadoc and renamed `theMap` to `rawInputRow`.